No colored output with custom Cargo subcommand alias

Running `cargo clippy` with `intellij-rust` produces colored terminal output for me under the current nightly of Cargo and `intellij-rust`.

With a custom Cargo subcommand alias, however, ‘Terminal’ output is colored tab within IntelliJ IDEA, but not when invoked as a ‘Run configuration’.

`.cargo/config`:
```toml
[alias]
stress_test = ["clippy", "-v", "--color=always", "--", ...]
```

This results in the command line:
`/usr/bin/rustup run nightly /usr/bin/cargo stress_test` ->
`/usr/bin/rustup run nightly /usr/bin/cargo clippy --color=always -v`

Setting a custom env var `TERM=xterm+256color` within the ‘Run configuration’ does not help.

See also: https://blog.jetbrains.com/idea/2017/07/intellij-idea-2017-2-is-here-smart-sleek-and-snappy/#comment-411103

I used ‘Import project from existing sources’ and accepted that `.idea` was to be overwritten. That did not work. Then I chose ‘Import project ...’ and then selecting Kotlin initially, and it did work, in the sense that I could browse the project. Finally I read up your `CONTRIBUTING.md`, but with this command things seem to have broken:
```sh
 ./gradlew :runIde

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':compileOnly'.
> Could not find lib/lz4-java-1.3.jar (com.jetbrains:ideaIC:2017.2.3).
  Searched in the following locations:
      file:/home/sanmai/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2017.2.3/6d3f27fdf86b1f9734e90dd1c42f46def3235a84/ideaIC-2017.2.3/com.jetbrains/ideaIC/2017.2.3/lib/lz4-java-1.3-2017.2.3.jar
      file:/home/sanmai/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2017.2.3/6d3f27fdf86b1f9734e90dd1c42f46def3235a84/ideaIC-2017.2.3/lib/lz4-java-1.3.jar
      file:/home/sanmai/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2017.2.3/33b40220debe52b3f69ad910233dd606234b198/lib/lz4-java-1.3-2017.2.3-[classifier].jar

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
3 actionable tasks: 1 executed, 2 up-to-date
```

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -43,6 +43,7 @@ oistein
 osialr
 pavel-v-chernykh
 pocket7878
+sanmai-NL
 Shiroy
 sirgl
 slavam2605

--- a/src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt
+++ b/src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt
@@ -103,8 +103,11 @@ class Cargo(
         val cmdLine = if (commandLine.channel == RustChannel.DEFAULT) {
             GeneralCommandLine(cargoExecutable)
         } else {
-            if (rustup == null) error("Channel '${commandLine.channel}' cannot be set explicitly because rustup is not available")
-            rustup.createRunCommandLine(commandLine.channel, cargoExecutable.toString())
+            if (rustup == null) {
+                error("Channel '${commandLine.channel}' cannot be set explicitly because rustup is not available")
+            } else {
+                GeneralCommandLine(cargoExecutable, "+${commandLine.channel}")
+            }
         }
 
         cmdLine
@@ -112,6 +115,8 @@ class Cargo(
             .withWorkDirectory(commandLine.workingDirectory ?: projectDirectory)
             .withParameters(commandLine.command)
             .withEnvironment(CargoConstants.RUSTC_ENV_VAR, rustExecutable.toString())
+            .withEnvironment("TERM", "ansi")
+            .withRedirectErrorStream(true)
 
         ProxyHelper().withProxyIfNeeded(cmdLine)
 
@@ -134,16 +139,13 @@ class Cargo(
             args
         }
 
-
-        // add colors
+        // Force colors
         if (colors
             && !SystemInfo.isWindows //BACKCOMPAT: remove windows check once termcolor'ed Cargo is stable
             && commandLine.command in COLOR_ACCEPTING_COMMANDS
             && args.none { it.startsWith("--color") }) {
 
             cmdLine
-                .withEnvironment("TERM", "ansi")
-                .withRedirectErrorStream(true)
                 .withParameters("--color=always") // Must come first in order not to corrupt the running program arguments
         }
 

